src/QMCDrivers/MCPopulation.cpp

```diff
@@ -44,8 +45,43 @@ MCPopulation::MCPopulation(int num_ranks,
 
 MCPopulation::~MCPopulation() = default;
 
+void MCPopulation::fissionHighMultiplicityWalkers()
+{
+  // we need to do this because spawnWalker changes walkers_ so we
+  // can't just iterate on that collection.
+  auto good_walkers = convertUPtrToRefVector(walkers_);
+  for (MCPWalker& good_walker : good_walkers)
+  {
+    int num_copies = static_cast<int>(good_walker.Multiplicity);
+    while (num_copies > 1)
+    {
+      auto walker_elements = spawnWalker();
+      // In the batched version walker ids are set when walkers are born and are unique,
+      // parent ids are set to the walker that provided the initial configuration.
+      // If the amplified walker was a transfer from another rank its copys get its ID
+      // as their parent, Not the walker id the received walker had on its original rank.
```

Comment:
If a walker A got transferred as B and then amplified to C.
```
B.parent_id = A.walker_id
C.parent_id = B.walker_id.
```
is this the scheme in use?
I kind of assume one generation (step) between walker_id and parent_id but for C there is 0 step between walker_id and parent_id. There is no issue for tracking but we cannot count the number steps anymore right?
Not a blocking issue but would like to make it clear.

Reply:
Yes, this requires a great deal less state than having fission of walkers that started on the rank follow one logic and fission of walkers that were transferred follow another.

Reply:
I think the real place to document this is `WalkerControl` and the manual but its potentially confusing enough that I felt like these notes might help in the future.

Reply:
I raised this question simply to capture my thoughts as was going through the PR not a direct issue of this PR.
For this function, its behavior has been documented clearly. My question was to the full scheme and I agree with you documentation for this behavior at WalkerControl or develop manual will be more appropriate.